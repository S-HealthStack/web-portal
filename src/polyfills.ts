import smoothScroll from 'smoothscroll-polyfill';
import _isNull from 'lodash/isNull';
import _isObject from 'lodash/isObject';
import browser from 'src/common/utils/browser';

type ArgumentTypes<F extends () => unknown> = F extends (...args: infer A) => unknown ? A : never;

// safari smooth scroll fix
if (browser.isSafari) {
  smoothScroll.polyfill();

  const prevScrollTo = window.scrollTo;

  // issue: https://github.com/iamdustan/smoothscroll/issues/161
  Object.assign(window, {
    scrollTo: ((...args) => {
      if (_isObject(args[0])) {
        args[0].left = _isNull(args[0].left) ? undefined : args[0].left;
        args[0].top = _isNull(args[0].top) ? undefined : args[0].top;
      } else {
        args[0] = _isNull(args[0]) ? undefined : args[0];
        args[1] = _isNull(args[1]) ? undefined : args[1];
      }

      prevScrollTo(...(args as ArgumentTypes<typeof window.scrollTo>));
    }) as typeof window.scrollTo,
  });
}
