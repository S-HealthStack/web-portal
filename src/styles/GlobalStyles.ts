import { createGlobalStyle } from 'styled-components';

export default createGlobalStyle`
  * {
    box-sizing: border-box;
  }

  html {
    font-size: 100%;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }

  html, body {
    height: 100%;
    width: 100%;
    margin: 0;
    position: fixed;
    overflow: hidden;
    overscroll-behavior-y: none;
    background: ${({ theme }) => theme.colors.background};
    user-select: none;
    -webkit-user-drag: none;
    font-family: Inter, sans-serif;
  }
  
  input,
  input:focus {
    ::-webkit-textfield-decoration-container {
      visibility: hidden;
    }
  }

  #root {
    height: 100%;
    width: 100%;
    display: flex;
    flex-direction: column;
    overflow: hidden;
  }
  
  *:focus,
  *:focus-visible {
    outline: none;
  }
  
  .recharts-surface {
    overflow: visible;
  }

`;
