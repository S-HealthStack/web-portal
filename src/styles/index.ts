export { default as theme } from './theme';
export {
  colorsFn as colors,
  typographyFn as typography,
  boxShadowFn as boxShadow,
  animationFn as animation,
} from './themeHelpers';
export { default as GlobalStyles } from './GlobalStyles';
export { px } from './utils';
