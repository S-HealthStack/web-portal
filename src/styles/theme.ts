import { css, DefaultTheme } from 'styled-components';
import { px } from './utils';

const colors = {
  // gray gradations
  black: '#000000',
  black40: 'rgba(0, 0, 0, 0.4)',
  black08: 'rgba(0, 0, 0, 0.08)',

  // light mode
  surface: '#FFFFFF',
  background: '#F7F8FA',
  onPrimary: '#FFFFFF',
  backgroundOnPrimary: '#FFFFFF',
  backgroundSurface: '#FFFFFF',
  backgroundOnSurface: '#2D2D2D',
  primary: '#4475E3',
  primaryLight: '#ECF1FC',
  primaryHovered: '#2F65E0',
  primary05: '#F6F8FE',
  primary10: '#ECF1FC',
  primary30: '#C7D6F7',
  primaryDisabled: '#B3C6F1',
  primaryLightPressed: '#E3EAFB',
  primaryLightFocused: '#C7D6F7',
  textPrimary: '#474747',
  primaryWhite: '#FFFFFF',
  disabled: '#DBDBDB',
  disabled20: 'rgba(219, 219, 219, 0.2)',
  onDisabled: '#A1A1A1',

  textPrimaryBlue: '#4475E3',
  textPrimaryDark: '#474747',
  textSecondaryGray: '#808080',
  textDisabled: '#838383',
  onSurface: '#2D2D2D',
  secondarySkyblue: '#00B0D7', // replace
  backgoundLight: '#F0F0F0',
  primaryBlueHovered: '#2F65E0',
  primaryBluePressed: '#2057D5',

  statusSuccess: '#2DC008',
  statusSuccess10: '#DCF2EA',
  statusSuccessText: '#00825D',

  statusError: '#FF3333',
  statusError10: '#FFEBEB',
  statusErrorText: '#D14343',

  statusWarning: '#F5BF00',
  statusWarning10: '#FFF7D0',
  statusWarningText: '#A36400',

  secondaryViolet: '#944ED7',
  secondaryVioletHovered: '#6A3799',
  secondaryViolet10: '#F4EDFB',

  secondaryTangerine: '#F39C18',
  secondaryTangerineHovered: '#CC8314',
  secondaryTangerine10: '#FEF5E8',

  secondarySkyBlue: '#00B0D7',
  secondarySkyBlueHovered: '#0092B3',
  secondarySkyBlue10: '#E5F7FB',

  secondaryGreen: '#1AD598',
  secondaryGreenHovered: '#16B37F',
  secondaryGreen10: '#E8FBF5',

  secondaryRed: '#FA5F4F',
  secondaryRedHovered: '#CC4E40',
  secondaryRed10: '#FFEFED',
};

export type SpecColorType = keyof typeof colors;

const weightToCssValue = {
  regular: 400,
  upperregular: 450,
  medium: 500,
  semibold: 600,
  bold: 700,
};

const fontSpec = (
  size: number,
  lineHeight: number | string,
  weight: keyof typeof weightToCssValue,
  fontFamily: string
) => css`
  font-family: ${fontFamily}, sans-serif;
  font-size: ${px(size)};
  line-height: ${typeof lineHeight === 'string' ? lineHeight : px(lineHeight)};
  font-weight: ${weightToCssValue[weight]};
  font-style: normal;
`;

const typography = {
  // Inter typeface
  labelRegular: fontSpec(10, 15, 'regular', 'Inter'),
  labelSemibold: fontSpec(10, 15, 'semibold', 'Inter'),

  bodyXSmallRegular: fontSpec(12, 18, 'regular', 'Inter'),
  bodyXSmallSemibold: fontSpec(12, 18, 'semibold', 'Inter'),
  bodySmallRegular: fontSpec(14, 21, 'regular', 'Inter'),
  bodySmallSemibold: fontSpec(14, 21, 'semibold', 'Inter'),
  bodyMediumRegular: fontSpec(16, 24, 'regular', 'Inter'),
  bodyMediumSemibold: fontSpec(16, 24, 'semibold', 'Inter'),
  bodyLargeRegular: fontSpec(18, 26, 'regular', 'Inter'),
  bodyLargeSemibold: fontSpec(18, 26, 'semibold', 'Inter'),

  headingXSmall: fontSpec(16, 21, 'semibold', 'Inter'),
  headingSmall: fontSpec(18, 23, 'semibold', 'Inter'),
  headingXMedium: fontSpec(18, 27, 'semibold', 'Inter'),
  headingXMediumRegular: fontSpec(18, 27, 'regular', 'Inter'),
  headingMedium: fontSpec(24, 32.2, 'semibold', 'Inter'),
  headingLargeSemibold: fontSpec(32, 42, 'semibold', 'Inter'),
  headingXLargeSemibold: fontSpec(48, 58, 'semibold', 'Inter'),
  headingXXLargeSemibold: fontSpec(60, 72, 'semibold', 'Inter'),

  // Fira Code typeface
  query14: fontSpec(14, '130%', 'upperregular', 'Fira Code'),
  query12: fontSpec(12, '150%', 'upperregular', 'Fira Code'),
};

const boxShadow = {
  card: `0 0 ${px(2)} ${px(0)} rgba(0,0,0,0.15)`,
  avatar: `0 ${px(8)} ${px(10)} ${px(2)} 0 rgba(0,0,0,0.12)`,
  studyAvatar: `0 ${px(15)} ${px(20)} rgba(180,180,180,0.3)`,
  previewScreen: `0 ${px(2)} ${px(4)} rgba(71,71,71,0.25)`,
};

const animation = {
  defaultTiming: 'cubic-bezier(0.45, 0.05, 0, 1)',
  rippleTiming: 'cubic-bezier(0.4, 0, 0.2, 1)',
};

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: typeof colors;
    typography: typeof typography;
    boxShadow: typeof boxShadow;
    animation: typeof animation;
  }
}

const theme: DefaultTheme = {
  colors,
  typography,
  boxShadow,
  animation,
};

export default theme;
