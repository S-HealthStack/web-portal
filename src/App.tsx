import React from 'react';
import { Provider } from 'react-redux';
import { Route, Switch } from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router';
import { ThemeProvider } from 'styled-components';
import { DndProvider } from 'react-dnd';
import { TouchBackend } from 'react-dnd-touch-backend';

import { store } from 'src/modules/store/store';
import { history, Path } from 'src/modules/navigation/store';
import PrivateRoute from 'src/modules/navigation/private-route/PrivateRoute';
import MainLayout from 'src/modules/main-layout/MainLayout';
import SignInScreen from 'src/modules/auth/signin/SignInScreen';
import AccountActivationScreen from 'src/modules/auth/signin/AccountActivationScreen';
import CreateStudyScreen from 'src/modules/auth/signin/CreateStudyScreen';
import { TooltipProvider, TooltipsList } from 'src/common/components/Tooltip';
import { theme, GlobalStyles } from 'src/styles';

import 'src/modules/auth/authProvider';

const App = () => (
  <ThemeProvider theme={theme}>
    <TooltipProvider>
      <DndProvider backend={TouchBackend} options={{ enableMouseEvents: true }}>
        <Provider store={store}>
          <GlobalStyles />
          <ConnectedRouter history={history}>
            <Switch>
              <Route path={Path.SignIn} component={SignInScreen} />
              <Route path={Path.AccountActivation} component={AccountActivationScreen} />
              <PrivateRoute path={Path.CreateStudy}>
                <CreateStudyScreen />
              </PrivateRoute>
              <PrivateRoute>
                <MainLayout />
              </PrivateRoute>
            </Switch>
          </ConnectedRouter>
        </Provider>
        <TooltipsList />
      </DndProvider>
    </TooltipProvider>
  </ThemeProvider>
);

export default App;
