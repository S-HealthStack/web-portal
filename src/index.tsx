import React from 'react';
import { createRoot } from 'react-dom/client';

import printAppVersion from 'src/common/utils/printAppVersion';

import './polyfills';
import App from './App';

printAppVersion();

const container = document.getElementById('root');
const root = createRoot(container as HTMLElement);

root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);
