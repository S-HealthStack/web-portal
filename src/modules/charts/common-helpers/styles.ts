import styled, { css } from 'styled-components';

import { colors, typography, px, animation } from 'src/styles';
import { LINE_CLASS_NAME } from '../common-components/Line';
import { AREA_CLASS_NAME } from '../common-components/Area';
import { DOT_CLASS_NAME } from '../common-components/Dot';

interface ContainerProps {
  width: number;
  height: number;
}

export const DOTS_CONTAINER_CLASS_NAME = 'dots';
export const X_AXIS_CLASS_NAME = 'xAxis';
export const Y_AXIS_CLASS_NAME = 'yAxis';
export const ZOOM_CLASS_NAME = 'zoom';

export const Container = styled.div.attrs<ContainerProps>(({ width, height }) => ({
  style: {
    width: px(width),
    height: px(height),
    position: 'relative',
  },
}))<ContainerProps>``;

export const TooltipContent = styled.div`
  display: flex;
  align-items: center;
`;

export const TooltipNameBlock = styled.div`
  display: flex;
  margin-right: ${px(8)};

  p {
    ${typography.labelSemibold};
    color: ${colors.surface};
    margin: 0;
  }
`;

export const TooltipDataBlock = styled.div`
  display: flex;
  flex-direction: column;

  p {
    ${typography.labelRegular};
    color: ${colors.surface};
    margin: 0;
  }
`;

export const TooltipColorPoint = styled.div<{ color: string }>`
  width: ${px(12)};
  height: ${px(12)};
  background-color: ${({ color }) => color};
  border-radius: 50%;
  margin-right: ${px(8)};
`;

export const brushOverlayStyles = css`
  fill: ${colors.primary};
  fill-opacity: 0.1;
  pointer-events: none;
`;

export const brushSelectionStyles = css`
  fill: ${colors.primary};
  fill-opacity: 1;
`;

export const brushHandleCommonStyles = css`
  stroke-width: ${px(1)};
  stroke: ${colors.primary};
  stroke-opacity: 0.3;
  fill: ${colors.onPrimary};
`;

export const StyledSvg = styled.svg<{
  focusWidth: number;
  $contextVisible: boolean;
}>`
  .${X_AXIS_CLASS_NAME} {
    .tick line {
      opacity: 0;
    }

    .tick {
      :first-child text {
        text-anchor: start;
      }
      :last-child text {
        text-anchor: end;
      }
    }
  }

  .${Y_AXIS_CLASS_NAME} {
    .tick line {
      stroke: ${colors.updBackgoundLight};
    }
  }

  .context {
    opacity: ${({ $contextVisible }) => ($contextVisible ? 1 : 0)};
    transition: opacity 0.3s ${animation.defaultTiming};
    .brushX {
      .overlay {
        ${brushOverlayStyles};
        height: ${px(4)};
      }
      .selection {
        ${brushSelectionStyles};
        height: ${px(22)};
        fill: url(#horizontalGradient);
      }
      .handle {
        ${brushHandleCommonStyles};
        height: ${px(22)};
        width: ${px(6)};
      }
    }
    .brushY {
      .overlay {
        ${brushOverlayStyles};
        width: ${px(4)};
      }
      .selection {
        ${brushSelectionStyles};
        width: ${px(22)};
        fill: url(#verticalGradient);
      }
      .handle {
        ${brushHandleCommonStyles};
        height: ${px(6)};
        width: ${px(22)};
      }
    }
  }

  .focus {
    cursor: move;
  }

  .${LINE_CLASS_NAME} {
    clip-path: url(#clip);
    transition: stroke-opacity 0.3s ${animation.defaultTiming};
  }

  .${AREA_CLASS_NAME} {
    clip-path: url(#clip);
    transition: fill-opacity 0.3s ${animation.defaultTiming};
  }

  .${DOTS_CONTAINER_CLASS_NAME} {
    clip-path: url(#clip);
    .${DOT_CLASS_NAME} {
      transition: fill-opacity 0.3s ${animation.defaultTiming};
    }
  }

  .zoom {
    fill: none;
    pointer-events: all;
  }
`;
