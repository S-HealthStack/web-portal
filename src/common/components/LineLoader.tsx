import React, { FC } from 'react';
import styled, { keyframes } from 'styled-components';

import { animation, colors, px } from 'src/styles';

const lineLoaderAnimation = keyframes`
  0% {
    width: 0;
    transform: translateX(0%);
  }
  7% {
    width: 10%;
    transform: translateX(0%);
  }
  40% {
    width: 60%;
  }
  50% {
    transform: translateX(85%);
  }
  75% {
    width: 100%;
  }
  100% {
    transform: translateX(100%);
  }
`;

export const LineLoaderRunner = styled.div`
  height: ${px(2)};
  background: ${colors.updPrimary};
`;

const LineLoaderContainer = styled.div`
  width: 100%;
  height: ${px(1)};
  background: ${colors.updPrimaryDisabled};
  overflow: hidden;

  ${LineLoaderRunner} {
    animation: 1.7s infinite ${lineLoaderAnimation} ${animation.defaultTiming};
  }
`;

const LineLoader: FC<React.HTMLAttributes<HTMLDivElement>> = (props) => (
  <LineLoaderContainer {...props}>
    <LineLoaderRunner />
  </LineLoaderContainer>
);

export default LineLoader;
