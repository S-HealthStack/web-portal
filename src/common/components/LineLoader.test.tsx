import React from 'react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import { px } from 'src/styles';
import theme from 'src/styles/theme';
import { ThemeProvider } from 'styled-components/';
import LineLoader from './LineLoader';

describe('LineLoader', () => {
  it('test line loader render', () => {
    const { baseElement, queryByTestId } = render(
      <ThemeProvider theme={theme}>
        <LineLoader />
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();

    const lineLoader = queryByTestId('line-loader') as Element;
    const runner = queryByTestId('runner') as Element;

    expect(lineLoader).toBeInTheDocument();
    expect(runner).toBeInTheDocument();

    expect(lineLoader).toHaveStyle(`height: ${px(1)}`);
    expect(lineLoader).toHaveStyle(`background: ${theme.colors.primaryDisabled}`);

    expect(runner).toHaveStyle(`height: ${px(2)}`);
    expect(runner).toHaveStyle(`background: ${theme.colors.primary}`);
  });

  it('[NEGATIVE] should render with excluding children', () => {
    const child = 'test';

    const { baseElement, queryByTestId } = render(
      <ThemeProvider theme={theme}>
        <LineLoader>{child}</LineLoader>
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();

    const lineLoader = queryByTestId('line-loader') as Element;
    const runner = queryByTestId('runner') as Element;

    expect(lineLoader).toBeInTheDocument();
    expect(runner).toBeInTheDocument();
    expect(lineLoader).not.toHaveTextContent(child);
  });
});
