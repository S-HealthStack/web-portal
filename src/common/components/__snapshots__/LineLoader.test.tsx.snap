// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineLoader [NEGATIVE] should render with excluding children 1`] = `
.c2 {
  height: 0.125rem;
  background: #4475E3;
}

.c0 {
  width: 100%;
  height: 0.0625rem;
  background: #B3C6F1;
  overflow: hidden;
}

.c0 .c1 {
  -webkit-animation: 1.7s infinite huZioT cubic-bezier(0.45,0.05,0,1);
  animation: 1.7s infinite huZioT cubic-bezier(0.45,0.05,0,1);
}

<body>
  <div>
    <div
      class="c0"
      data-testid="line-loader"
    >
      <div
        class="c1 c2"
        data-testid="runner"
      />
    </div>
  </div>
</body>
`;

exports[`LineLoader test line loader render 1`] = `
.c2 {
  height: 0.125rem;
  background: #4475E3;
}

.c0 {
  width: 100%;
  height: 0.0625rem;
  background: #B3C6F1;
  overflow: hidden;
}

.c0 .c1 {
  -webkit-animation: 1.7s infinite huZioT cubic-bezier(0.45,0.05,0,1);
  animation: 1.7s infinite huZioT cubic-bezier(0.45,0.05,0,1);
}

<body>
  <div>
    <div
      class="c0"
      data-testid="line-loader"
    >
      <div
        class="c1 c2"
        data-testid="runner"
      />
    </div>
  </div>
</body>
`;
